### PR TITLE
Dt 2330 use booking alerts v2

### PR DESCRIPTION
**Pull request by andymarke**
Old version had possible vulnerabilities in, so new version acts the same but with a different interface the the modern paging model

### CODE REVIEW

File: backend/api/prisonApi.ts [hunk 1]
```diff
@@ -96,9 +96,11 @@ export const prisonApiFactory = (client) => {
...
 
-  const getAlertsForBooking = (context, { bookingId, query }, headers) => {
-    contextProperties.setCustomRequestHeaders(context, headers)
-    return get(context, `/api/bookings/${bookingId}/alerts${query}`)
+  const getAlertsForBookingV2 = (context, { bookingId, alertType, from, to, alertStatus, page, sort }) => {
+    return get(
+      context,
+      `/api/bookings/${bookingId}/alerts/v2?alertType=${alertType}&from=${from}&to=${to}&alertStatus=${alertStatus}&page=${page}&sort=${sort}`
+    )
   }
```
Comment (andymarke) [inline backend/api/prisonApi.ts:104]: new API is happy to send empty values for any filter not needed

File: backend/tests/prisonerAlerts.test.ts [hunk 2]
```diff
@@ -111,8 +140,6 @@ describe('prisoner alerts', () => {
       active: 'ACTIVE',
     }
 
-    res.locals.responseHeaders['total-records'] = 1
-
```
Comment (andymarke) [inline backend/tests/prisonerAlerts.test.ts:115]: this was cheekily pretending there was a single record, when the mock actually has none, so removed and adjusted the expectation below to "0" records

File: backend/tests/prisonerAlerts.test.ts [hunk 3]
```diff
@@ -145,7 +174,7 @@ describe('prisoner alerts', () => {
           { text: 'Child Communication Measures', value: 'C' },
           { text: 'Social Care', value: 'A' },
         ],
-        totalAlerts: 1,
+        totalAlerts: 0,
         alertType: 'X',
```
Comment (andymarke) [inline backend/tests/prisonerAlerts.test.ts:178]: see comment above

**Review by JamieDraperUK — APPROVED** [review]
LGTM

**Pull request merged**